### ISSUE TITLE
Tests are reported as terminated under the debugger

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#4276]
## Environment

* **IntelliJ Rust plugin version:** 0.2.103.2131-192
* **Rust toolchain version:** rustc 1.36.0 (a53f9df32 2019-07-03)
* **IDE name and version:** 2019.2 CLion 192.5728.100
* **Operating system:** macOS 10.14.6

## Problem description

There are multiple problems with debugging tests:

1. Tests are reported as "Terminated"
2. One test case is permanently stuck in the "In Progress" state
3. Test hierarchy tree is broken

<img width="626" alt="Screen Shot 2019-08-19 at 13 42 17" src="https://user-images.githubusercontent.com/2500320/63259579-7b81fa00-c287-11e9-823e-86affb3fe9f7.png">

## Steps to reproduce

Debug the following module by clicking on the gutter icon next to `mod` declaration:

```rust
#[cfg(test)]
mod tests {
    #[test]
    fn test_sample() {
        assert_eq!(1 + 1, 2);
    }

    #[test]
    fn test_sample2() {
        assert_eq!(1 + 1, 2);
    }
}
```

N.B. it works fine under Run.
